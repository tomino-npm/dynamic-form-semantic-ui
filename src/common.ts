import { FormElement, DataSet } from '@tomino/dynamic-form';
import { initCss } from '@tomino/toolbelt';

export type SignatureType = {
  comment: string;
  signature: string;
  verifiedState: 'Pending' | 'Verified' | 'Rejected';
  rejected?: boolean;
  uid: string;
  name: string;
  date: Date;
  setValue?(name: string, value: any): any;
};

export type SignatureHandlers = {
  validateForm(): boolean;
  verifySignature(uid: string, signature: string): Promise<string>;
  sign(
    source: string,
    reject: boolean,
    password: string,
    reason: string,
    submit: boolean
  ): Promise<SignatureType>;
  signatureFont(): string;
};

export type FormControlProps = {
  formControl: FormElement;
  owner: DataSet;
  handlers?: { [index: string]: any };
  readOnly: boolean;
};

export const { css, renderCss } = initCss('form-');
