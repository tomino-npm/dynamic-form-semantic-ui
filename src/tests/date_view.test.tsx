import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { create } from './form_query_data';
import { JSONSchema, FormDefinition, FormModel } from '@tomino/dynamic-form';
import { TestComponent } from './common';

const schema: JSONSchema = {
  type: 'object',
  properties: {
    date: {
      type: 'string',
      format: 'date'
    },
    range: {
      type: 'string'
    }
  }
};

const formDefinition: FormDefinition = create.form({
  elements: [
    create.formElement({
      row: 0,
      column: 0,
      width: 4,
      control: 'Date',
      label: 'Date',
      source: 'date'
    }),
    create.formElement({
      row: 0,
      column: 4,
      width: 4,
      control: 'Time',
      label: 'Time',
      source: 'date'
    }),
    create.formElement({
      row: 0,
      column: 8,
      width: 4,
      control: 'DateTime',
      label: 'Date Time',
      source: 'date'
    }),
    create.formElement({
      row: 0,
      column: 8,
      width: 4,
      control: 'DateRange',
      label: 'Date Range',
      source: 'range'
    })
  ]
});

function component() {
  const form = new FormModel(formDefinition, schema, {});

  // just another notation
  return <TestComponent form={form} />;
}

describe('DateView', () => {
  it('renders', function() {
    const wrapper = renderer.create(component());
    expect(wrapper).toMatchSnapshot();
  });
  return {
    component
  };
});
