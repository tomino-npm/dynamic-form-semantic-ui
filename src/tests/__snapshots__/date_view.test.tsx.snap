// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateView renders 1`] = `
<div
  className="ui segment ui form"
>
  <div
    className="ui dividing header"
  >
    Standard
  </div>
  <div
    className="ui form "
  >
    <div
      className="fields"
    >
      <div
        className="four wide field"
      >
        <label
          htmlFor="date"
        >
          Date
        </label>
        <div>
          <div
            className="field"
          >
            <div
              className="ui left icon input"
            >
              <i
                aria-hidden="true"
                className="calendar icon"
                onClick={[Function]}
              />
              <input
                name="date"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                tabIndex="0"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="four wide field"
      >
        <label
          htmlFor="date"
        >
          Time
        </label>
        <div>
          <div
            className="field"
          >
            <div
              className="ui left icon input"
            >
              <i
                aria-hidden="true"
                className="time icon"
                onClick={[Function]}
              />
              <input
                name="date"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                tabIndex="0"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="four wide field"
      >
        <label
          htmlFor="date"
        >
          Date Time
        </label>
        <div>
          <div
            className="field"
          >
            <div
              className="ui left icon input"
            >
              <i
                aria-hidden="true"
                className="calendar icon"
                onClick={[Function]}
              />
              <input
                name="date"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                tabIndex="0"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="four wide field"
      >
        <label
          htmlFor="range"
        >
          Date Range
        </label>
        <div>
          <div
            className="field"
          >
            <div
              className="ui left icon input"
            >
              <i
                aria-hidden="true"
                className="calendar icon"
                onClick={[Function]}
              />
              <input
                name="range"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                tabIndex="0"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui dividing header"
  >
    Readonly
  </div>
  <div
    className="ui form form-hjcaghiia"
  >
    <div
      className="fields"
    >
      <div
        className="four wide field"
      >
        <label
          htmlFor="date"
        >
          Date
        </label>
        <div>
          <div
            className="field"
          >
            <div
              className="ui left icon input"
            >
              <i
                aria-hidden="true"
                className="calendar icon"
                onClick={[Function]}
              />
              <input
                name="date"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                tabIndex="0"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="four wide field"
      >
        <label
          htmlFor="date"
        >
          Time
        </label>
        <div>
          <div
            className="field"
          >
            <div
              className="ui left icon input"
            >
              <i
                aria-hidden="true"
                className="time icon"
                onClick={[Function]}
              />
              <input
                name="date"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                tabIndex="0"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="four wide field"
      >
        <label
          htmlFor="date"
        >
          Date Time
        </label>
        <div>
          <div
            className="field"
          >
            <div
              className="ui left icon input"
            >
              <i
                aria-hidden="true"
                className="calendar icon"
                onClick={[Function]}
              />
              <input
                name="date"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                tabIndex="0"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="four wide field"
      >
        <label
          htmlFor="range"
        >
          Date Range
        </label>
        <div>
          <div
            className="field"
          >
            <div
              className="ui left icon input"
            >
              <i
                aria-hidden="true"
                className="calendar icon"
                onClick={[Function]}
              />
              <input
                name="range"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                tabIndex="0"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
